tests/test_file_event_actions.py

```diff
@@ -0,0 +1,626 @@
+    ):
+        eq_param = {
+            "username": "test@example.com",
+            "file_hash": "6849f4e9b2ee9d45052145d8e25d7b99",
+            "file_name": "filename.txt",
+            "file_path": "/some/path/to/a/file",
+            "file_category": "Spreadsheet",
+            "hostname": "MSEDGEWIN10",
+            "private_ip": "65.29.153.33",
+            "public_ip": "127.0.0.1",
+            "exposure_type": "ApplicationRead",
+            "process_name": "/Device/HarddiskVolume1/Program Files (x86)/Microsoft/Edge/Application/msedge.exe",
+            "url": "https://drive.google.com/drive/my-drive",
+            "window_title": "My Drive - Google Drive and 1 more page - Profile 1 - Microsoft​ Edge",
+        }
+
+        date_param = {
+            "start_date": "2031-01-01",
+            "end_date": "2031-06-01",
+        }
+
+        param = {}
+        param.update(eq_param)
+        param.update(date_param)
+
+        connector = _create_run_query_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+
+        actual_query = mock_py42_client.securitydata.search_file_events.call_args[0][0]
+        query_json = json.loads(str(actual_query))
+
+        for key in eq_param:
+            check_filter(query_json, "IS", eq_param[key])
+
+        check_filter(query_json, "ON_OR_AFTER", "2031-01-01T00:00:00.000Z")
+        check_filter(query_json, "ON_OR_BEFORE", "2031-06-01T00:00:00.000Z")
+
+        assert_success(connector)
+
+    def test_handle_action_when_run_query_and_all_params_missing_sets_error_message(
+        self, mocker, mock_py42_client
+    ):
+        param = {}
+        connector = _create_run_query_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        expected_message = (
+            "Code42: Must supply a search term when calling action 'run_query'."
+        )
+        assert_fail_message(connector, expected_message)
+
+    def test_handle_action_when_run_query_and_only_start_date_set_calls_search(
+        self, mocker, mock_py42_client
+    ):
+        param = {"start_date": "2031-01-01"}
+        connector = _create_run_query_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        assert_success(connector)
+
+    def test_handle_action_when_run_query_and_only_end_date_set_calls_search(
+        self, mocker, mock_py42_client
+    ):
+        param = {"end_date": "2031-01-01"}
+        connector = _create_run_query_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        assert_success(connector)
+
+    def test_handle_action_when_run_query_and_date_format_invalid_sets_error_status(
+        self, mocker, mock_py42_client
+    ):
+        param = {
+            "start_date": "2031-31-12",
+            "end_date": "2032-31-01",
+        }
+        connector = _create_run_query_connector(mocker, mock_py42_client)
+        connector.handle_action(param)
+        assert_fail(connector)
```

Comment:
I am curious what the failure message looks like in this case

Reply:
added comment
Did this rather than `assert_failure_message` since the error message in this case comes from date util or something.. it looks decent though and explains the problem well enough, which was what I wanted to make sure of.